tests/ui/deprecated.rs

```diff
@@ -3,5 +3,6 @@
 #[warn(unstable_as_slice)]
 #[warn(unstable_as_mut_slice)]
 #[warn(misaligned_transmute)]
+#[warn(unused_collect)]
```

Comment:
I think we should move this test to `deprecated_old.rs` and create a new test file for tool lints deprecation.